pkg/plugin/installer/installer.go

```diff
@@ -0,0 +1,70 @@
+package installer
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/Azure/draft/pkg/draft/draftpath"
+)
+
+// special thanks to the Kubernets Helm plugin installer pkg
+
+// ErrMissingMetadata indicates that plugin.yaml is missing.
+var ErrMissingMetadata = errors.New("plugin metadata (plugin.yaml) missing")
+
+// Debug enables verbose output.
+var Debug bool
+
+// Installer provides an interface for installing client plugins.
+type Installer interface {
+	// Install adds a plugin to a path
+	Install() error
+	// Path is the directory of the installed plugin.
+	Path() string
+}
+
+// Install installs a plugin to $DRAFT_HOME
+func Install(i Installer) error {
+	if _, pathErr := os.Stat(path.Dir(i.Path())); os.IsNotExist(pathErr) {
+
+		return errors.New(`plugin home "$DRAFT_HOME/plugins" does not exist`)
+	}
+
+	if _, pathErr := os.Stat(i.Path()); !os.IsNotExist(pathErr) {
+		return errors.New("plugin already exists")
+	}
+
+	return i.Install()
+}
+
+// New determines and returns the correct Installer for the given source
+func New(source, version string, home draftpath.Home) (Installer, error) {
+	if isLocalReference(source) {
+		return NewLocalInstaller(source, home)
+	}
+
+	return NewVCSInstaller(source, version, home)
+}
+
+func debug(format string, args ...interface{}) {
+	if Debug {
+		format = fmt.Sprintf("[debug] %s\n", format)
+		fmt.Printf(format, args...)
+	}
+}
+
+// isLocalReference checks if the source exists on the filesystem.
+func isLocalReference(source string) bool {
+	//TODO: can user install a local plugin using "."?
```

Comment:
I would say no, it has to be a path for now (even if the path is just `./`. We should document that.